### PR TITLE
fixed filter styling and added js and non-js filter form submit

### PR DESCRIPTION
**Pull request by IncredibleMrTim**
### What

- updated dp-renderer to version to point at new version containing StringArrayContains function
- added functionality from dp-design-system to allow filters to be submitted via js and non-js
- implemented the checkbox component from the design system

### How to review

- pull the `dp-design-system` branch `feature/search-bar` and run locally
- run `dp-frontend-search-controller`
- enter a search and press enter
- the filters down the left hand side of the page should be displayed as expanded
- selecting a child filter should filter the page
- turn of JS, reload the page and the filter button should be visible allowing you to still filter the results
- filter options should persist after the page results refresh or the page is forcefully refreshed
### Who can review

Describe who worked on the changes, so that other people can review.

Anyone

### CODE REVIEW

File: assets/templates/partials/filter_old.tmpl [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{{ $filter := .Data.Filter }}
+{{ $response := .Data.Response }}
+{{ $lang := .Language }}
+<form id="form" class="js-auto-submit__form" action="">
+    <div class="col col--md-12 col--lg-18 margin-bottom">
+        <div class="margin-bottom-md--2 flush-col js-mobile-filters">
+
```
Comment (AnnaEmson) [inline assets/templates/partials/filter_old.tmpl:7]: worth removing this empty line as you should not have a space between html elements

File: assets/templates/partials/filter_old.tmpl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+{{ $lang := .Language }}
+<form id="form" class="js-auto-submit__form" action="">
+    <div class="col col--md-12 col--lg-18 margin-bottom">
+        <div class="margin-bottom-md--2 flush-col js-mobile-filters">
+            
+            <div class="background--gallery padding-top-md--2 padding-right-md--1 padding-bottom-md--4 padding-left-sm--1 padding-left-md--1 flush js-mobile-filters__title">
+                <h3 class="inline-block flush">{{ localise "RefineResults" $lang 4 }}</h3>
+                <a href="/search{{ if .Data.Query }}?q={{ .Data.Query}}{{end}}" id="clear-search" class="float-right">{{ localise "ClearAll" $lang 1 }}</a>
+            </div>
+
```
Comment (AnnaEmson) [inline assets/templates/partials/filter_old.tmpl:12]: the same here

Comment (IncredibleMrTim) [reply assets/templates/partials/filter_old.tmpl]: fixed

File: assets/templates/partials/filter.tmpl [hunk 3]
```diff
@@ -1,50 +1,85 @@
...
+                                                <span class="ons-checkboxes__item ons-checkboxes__item--no-border">
+                                                    <span class="ons-checkbox ons-checkbox--no-border child-filter">
+                                                        {{ $id := (print $content.Type) }}
+                                                        <input id="{{ $id }}"
+                                                               class="js-auto-submit__input ons-checkbox__input ons-js-checkbox"
+                                                               type="checkbox" name="filter"
+                                                               {{ range $filter }}{{ if eq . $content.Type }}checked{{end}}{{end}}
+                                                               value="{{ $id }}"
+                                                        />
+
```
[the remark this answers is not in the document]
Comment (IncredibleMrTim) [reply assets/templates/partials/filter.tmpl]: fixed

File: assets/templates/partials/results.tmpl [hunk 4]
```diff
@@ -3,69 +3,70 @@
...
+                    <li>search again using different words</li>
+                </ul>
+            </div>
+            <div class="search-results margin-top-md--2">
+                <ul class="list--neutral flush">
+                </ul>
+            </div>
+        {{end}}
+    </h2>
 </div>
```
Comment (NoelleDL) [inline assets/templates/partials/results.tmpl:72]: New line needed.

File: assets/templates/partials/summary.tmpl [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                        <a class="underline-link" href="/search?q={{ $suggestion }}">{{ $suggestion }}</a>
+                        {{ if notLastItem $len $i }}
+                            or
+                        {{ end }}
+                    {{ end }}
+                {{ end }}
+            {{ end }}
+        </span>
+    </div>
+</div>
```
Comment (NoelleDL) [inline assets/templates/partials/summary.tmpl:36]: New line needed.

File: assets/templates/search.tmpl [hunk 6]
```diff
@@ -1,11 +1,24 @@
...
+            {{ template "partials/filter" . }}
+        </section>
+        <section class="col col--md-34 col--lg-40 margin-left-md--1" aria-label="Search results">
+            {{ template "partials/department" . }}
+            {{ template "partials/results" . }}
+            {{ template "partials/pagination" . }}
+        </section>
     </div>
-</div>
+</div>
```
Comment (NoelleDL) [inline assets/templates/search.tmpl:24]: New line needed.

File: assets/templates/partials/results.tmpl (hunk 4 above)
Comment (IncredibleMrTim) [reply assets/templates/partials/results.tmpl]: fixed

File: assets/templates/search.tmpl (hunk 6 above)
Comment (IncredibleMrTim) [reply assets/templates/search.tmpl]: fixed

File: assets/templates/partials/summary.tmpl (hunk 5 above)
Comment (IncredibleMrTim) [reply assets/templates/partials/summary.tmpl]: fixed

**Review by NoelleDL — APPROVED** [review]

**Pull request merged**